### DISCUSSION TITLE
chatglm3-6b 模型response内有多个<|user|> <|assistant|>

### DISCUSSION THREAD

**Discussion by forrestjgq** [zai-org/ChatGLM3#514] [Q&A] [answered]
Hi:
我在用triton+trtllm 部署chatglm3-6b模型，发现一个问题，下面是我的模型prompt：
```
<|system|>你是一位智能AI助手<|user|>谁赢了2020年世界杯?<|assistant|>法国队赢了<|user|>这次是哪里举办的？<|assistant|>
```
模型回复：
```
这次世界杯是在卡塔尔举办的。<|user|>法国队的队员们都来自哪些国家?<|assistant|>法国队的队员来自各个国家，但主要来自欧洲。
```

我期待的是`这次世界杯是在卡塔尔举办的。`，但是后面有额外的user和assistant段，这个是使用问题吗？如何避免这样的问题？

**Comment by HillkingF** [comment]
请问这个问题您解决了不？我也遇到了这种情况

**Reply by forrestjgq** [reply]
没有解决啊，等着官方给回复呢

**Reply by zRzRzRzRzRzRzR** [reply]
这是官方的Special token，确保你推理的时候完全复现了官方的实现脚本和特殊token截断

**Comment by forrestjgq** [comment]
> 请问这个问题您解决了不？我也遇到了这种情况
@HillkingF 我看了官方代码，里面把end id和<|user|>, <|observation|>都当作eos

@zRzRzRzRzRzRzR 我用tokenizer的`get_command("<|user|>")`获得的token是对的，但是直接`tokenizer("<|user|>")`会出来3个token，这个是什么用法错误吗？如果这样不可以的话，我去做这样一个prompt然后tokenize岂非拿到的是错误的id？
```
<|system|>你是一位智能AI助手<|user|>谁赢了2020年世界杯?<|assistant|>法国队赢了<|user|>这次是哪里举办的？<|assistant|>
```

**Reply by zRzRzRzRzRzRzR** [reply, marked as answer]
你使用防注入了吗，Huggingface推理的防注入要开才会得到正确id

**Reply by ruyiluo** [reply]
请问如何实现防注入呢？

**Reply by zRzRzRzRzRzRzR** [reply]
skip_special_tokens=True

**Reply by ruyiluo** [reply]
好的，感谢
